src/users_db.rs

```diff
@@ -356,28 +355,25 @@ impl UsersDb {
                 );
                 try!(stmt.query(&[&id]))
             },
-            ReadFilter::Name(name) => {
+            ReadFilter::Name(ref name) => {
                 stmt = try!(
                     self.connection.prepare("SELECT * FROM users WHERE name=$1")
                 );
                 try!(stmt.query(&[&escape(&name)]))
             },
-            ReadFilter::Email(email) => {
+            ReadFilter::Email(ref email) => {
                 stmt = try!(
                     self.connection.prepare("SELECT * FROM users WHERE email=$1")
                 );
-                try!(stmt.query(&[&escape(&email)]))
+                try!(stmt.query(&[&escape(email)]))
             },
-            ReadFilter::Credentials(user, password) => {
-                let mut md5 = Md5::new();
-                md5.input_str(&escape(&password));
-                let password = md5.result_str();
+            ReadFilter::Credentials(ref user, _) => {
```

Comment:
Not sure what you mean. The ````ReadFilter::Credentials```` is defined as carrying a pair of String (user / password). Here I don't need the password since I can't compare it from the SELECT statement as I need the salt to do that. But below when we convert the ````Rows```` to ````User```` I perform the filter on the password.